src/main/java/io/vertx/core/http/impl/ServerWebSocketImpl.java

```diff
@@ -135,15 +149,18 @@ private void connect() {
   // Connect if not already connected
   void connectNow() {
     synchronized (conn) {
-      if (!connected && !rejected) {
+      if (!connected && !isRejected()) {
         connect();
       }
     }
   }
 
-  boolean isRejected() {
-    synchronized (conn) {
-      return rejected;
-    }
+  synchronized boolean isRejected() {
```

Comment:
`synchronized(conn)` rather than `synchronized(this)` is likely what you expected here